services/worker-manager/src/providers/azure.js

```diff
@@ -52,13 +52,14 @@ function generateAdminPassword() {
 
 function workerConfigWithSecrets(cfg) {
   assert(_.has(cfg, 'osProfile'));
+  let newCfg = _.cloneDeep(cfg);
```

Comment:
This bit is thrown in to ensure that we don't save `adminPassword` and `adminUsername` when we save parts of `cfg` into the VM `providerData.config`